builtin/game/register.lua

```diff
@@ -407,6 +407,29 @@ local function make_registration_wrap(reg_fn_name, clear_fn_name)
 	return list
 end
 
+core.registered_on_player_hpchanges = { modifiers = { }, loggers = { } }
+function core.registered_on_player_hpchange(player, hp_change)
+	for i, func in ipairs(core.registered_on_player_hpchanges.modifiers) do
+		hp_change = func(player, hp_change)
+		if hp_change == nil then
```

Comment:
Perhaps you make a real check for numbers here? `if type(hp_change) ~= "number" then`